### PR TITLE
Exercise rational numbers

### PR DESCRIPTION
**Pull request by pclausen**
Exercise rational numbers

### CODE REVIEW

File: testlib/TesterMain.f90 [hunk 1]
```diff
@@ -149,6 +150,21 @@ subroutine assert_equal_dble(e_dble,i_dble,msg)
     endif
   end subroutine
 
+  !------------------------------------------------------------------
+  subroutine assert_equal_real(e_real,i_real,msg)
```
Comment (hiljusti) [inline testlib/TesterMain.f90:154]: Wow - really appreciate the work you've been putting in here!

File: exercises/practice/rational-numbers/.meta/example.f90 [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  function div(r1,r2)
+    integer,dimension(2) :: div, r1,r2
+    ! r₁ / r₂ = (a₁ * b₂) / (a₂ * b₁)
+    div(1) = r1(1)*r2(2)
+    div(2) = r1(2)*r2(1)
+    div = reduce(div)
+  end function
+
+  function rabs(r1)
```
Comment (hiljusti) [inline exercises/practice/rational-numbers/.meta/example.f90:38]: What do you think of longer function names like `rational_abs`?

File: exercises/practice/rational-numbers/.meta/example.f90 [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    integer,dimension(2) :: rabs, r1
+    rabs(1) = abs(r1(1))
+    rabs(2) = abs(r1(2))
+  end function
+
+
+  function exprational(r1, ex)
```
Comment (hiljusti) [inline exercises/practice/rational-numbers/.meta/example.f90:45]: What do you think of longer function names like `rational_to_pow`?

File: exercises/practice/rational-numbers/.meta/example.f90 [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    integer,dimension(2) :: exprational, r1
+    integer :: ex
+    !(a1 * b2 + a2 * b1) / (b1 * b2)
+    exprational(1) = r1(1)**ex
+    exprational(2) = r1(2)**ex
+  end function
+
+  
+  function expreal(ex,r1)
```
Comment (hiljusti) [inline exercises/practice/rational-numbers/.meta/example.f90:54]: What do you think of a longer function name like `to_rational_pow`?

**Review by hiljusti — CHANGES_REQUESTED** [review]

File: exercises/practice/rational-numbers/rational_numbers.f90 [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+module rational_numbers
+  implicit none
+contains
+
+
+end module
```
[the remark this answers is not in the document]
Comment (pclausen) [reply exercises/practice/rational-numbers/rational_numbers.f90:7]: Adding stub functions

File: exercises/practice/rational-numbers/.meta/example.f90 (hunk 2 above)
Comment (pclausen) [reply exercises/practice/rational-numbers/.meta/example.f90:38]: I used the names from the problem-sepcification but yeah, I am fine with another name

File: exercises/practice/rational-numbers/.meta/example.f90 (hunk 3 above)
Comment (pclausen) [reply exercises/practice/rational-numbers/.meta/example.f90:45]: OK

**Comment by pclausen** [comment]
@hiljusti  Thanks for the review. I integrated the changes you suggested.

**Comment by hiljusti** [comment]
Awesome, thanks for taking a look - btw feel free to take any of my suggestions as just suggestions. I think you're more of an expert here than I am!

**Review by hiljusti — APPROVED** [review]

**Pull request merged**